examples/virtual_file_server.rb

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/ruby
+
+require 'bundler/setup'
+require 'optparse'
+require 'ruby_smb'
+require 'ruby_smb/gss/provider/ntlm'
+
+# we just need *a* default encoding to handle the strings from the NTLM messages
+Encoding.default_internal = 'UTF-8' if Encoding.default_internal.nil?
```

Comment:
Well, this is admittedly super sketch. I wouldn't do this in the library to avoid affecting downstream consumers. The existing [file_server](https://github.com/rapid7/ruby_smb/blob/e98f2633ce08ccf1ee5239868202b80eeda7e655/examples/file_server.rb#L9) does the exact same thing. Also, if I remove the line, the server crashes which is no good. I think the best case might be to find the instances where this was necessary and do the `some_input.encode(''.encoding)` trick, which to my knowledge is still the only way to coerce a string of an arbitrary encoding into one that's compatible with comparison.